plugins/system_upgrade.py

```diff
@@ -380,6 +381,11 @@ def pre_configure_download(self):
         # only download subcommand accepts --destdir command line option
         self.base.conf.cachedir = DEFAULT_DATADIR
         self.base.conf.destdir = self.opts.destdir if self.opts.destdir else None
+        if 'offline-distrosync' == self.opts.command and not self.opts.distro_sync:
+            raise CliError(
+                _("Command 'offline-distrosync' cannot be used with --no-downgrade option"))
```

Comment:
Uhh, what `--no-downgrade` option?

Reply:
--no-downgrade option is not a new option. dnf system-upgrade --no-downgrade option, will use goal.upgrade() instead of goal.distrosync.